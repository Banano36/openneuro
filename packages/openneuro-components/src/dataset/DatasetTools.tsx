import React from 'react'
import { DatasetToolButton } from './DatasetToolButton'
import styled from '@emotion/styled'

export const DatasetToolStyle = styled.span`
  display: flex;
  justify-content: flex-start;
  flex-wrap: wrap;
`

export interface DatasetToolsProps {
  hasEdit: boolean
  isPublic: boolean
  datasetId: string
  snapshotId?: string
  isAdmin: boolean
  isDatasetAdmin: boolean
  hasSnapshot?: boolean
}

export const DatasetTools = ({
  hasEdit,
  isPublic,
  datasetId,
  snapshotId,
  isAdmin,
  hasSnapshot,
  isDatasetAdmin,
}: DatasetToolsProps) => {
  const isSnapshot = snapshotId
  return (
    <DatasetToolStyle>
      {hasEdit && !isPublic && !isSnapshot && (
        <>
          {hasSnapshot ? (
            <DatasetToolButton
              tooltip="Publicize the dataset"
              path={`/datasets/${datasetId}/publish`}
              icon="fa-globe"
              label="Publish"
            />
          ) : (
            <DatasetToolButton
              tooltip="Create a version to publish"
              path={`/datasets/${datasetId}/snapshot`}
              icon="fa-globe"
              label="Publish"
            />
          )}
        </>
      )}
      {hasEdit && !isSnapshot && (
        <DatasetToolButton
          tooltip="Share this dataset with collaborators"
          path={`/datasets/${datasetId}/share`}
          icon="fa-user"
          label="Share"
        />
      )}
      {hasEdit && isSnapshot && (
        <DatasetToolButton
          tooltip="View the Dataset Draft"
          path={`/datasets/${datasetId}`}
          icon="fa-pencil"
          label="View Draft"
        />
      )}
      {hasEdit && !isSnapshot && (
        <DatasetToolButton
          tooltip="Create a new version of the dataset"
          path={`/datasets/${datasetId}/snapshot`}
          icon="fa-camera"
          label="Versioning"
        />
      )}
      {isAdmin && !isSnapshot && (
        <DatasetToolButton
          tooltip="Admin Datalad Tools"
          path={`/datasets/${datasetId}/admin`}
          icon="fa-magic"
          label="Admin"
        />
      )}
      <DatasetToolButton
        tooltip="How to Download"
        path={
          snapshotId
            ? `/datasets/${datasetId}/versions/${snapshotId}/download`
            : `/datasets/${datasetId}/download`
        }
        icon="fa-download"
        label="Download"
      />
      <DatasetToolButton
        tooltip={
          hasEdit
            ? 'A form to describe your dataset (helps colleagues discover your dataset)'
            : 'View the dataset metadata'
        }
        path={
          snapshotId
            ? `/datasets/${datasetId}/versions/${snapshotId}/metadata`
            : `/datasets/${datasetId}/metadata`
        }
        icon="fa-file-code"
        label="Metadata"
      />
      {isDatasetAdmin && !isSnapshot && (
        <DatasetToolButton
          tooltip="Remove your dataset from OpenNeuro"
          path={`/datasets/${datasetId}/delete`}
          icon="fa-trash"
          label="Delete"
        />
      )}
      {hasEdit && isSnapshot && (
        <DatasetToolButton
          tooltip="Flag this version as deprecated"
          path={`/datasets/${datasetId}/versions/${snapshotId}/deprecate`}
          icon="fa-remove"
          label="Deprecate Version"
        />
      )}
    </DatasetToolStyle>
  )
}
