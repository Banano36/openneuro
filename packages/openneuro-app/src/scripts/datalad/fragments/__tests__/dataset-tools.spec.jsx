import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import DatasetTools from '../dataset-tools.jsx'

describe('DatasetTools component', () => {
  it('renders tools with common props', () => {
    const wrapper = render(
      <MemoryRouter initialEntries={[`/datasets/ds000113`]}>
        <DatasetTools
          dataset={{
            id: 'ds001',
            public: false,
            following: true,
            starred: false,
            snapshots: [],
          }}
        />
      </MemoryRouter>,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('renders tools for public datasets', () => {
    const wrapper = render(
      <MemoryRouter initialEntries={[`/datasets/ds000113`]}>
        <DatasetTools
          dataset={{
            id: 'ds001',
            public: true,
            following: true,
            starred: false,
            snapshots: [],
          }}
        />
      </MemoryRouter>,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('renders tools for public snapshots', () => {
    const wrapper = render(
      <MemoryRouter initialEntries={[`/datasets/ds000113/versions/1.3.0`]}>
        <DatasetTools
          dataset={{
            id: 'ds001',
            public: true,
            following: true,
            starred: false,
            snapshots: [{ id: 'ds000113:1.3.0' }],
          }}
        />
      </MemoryRouter>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
