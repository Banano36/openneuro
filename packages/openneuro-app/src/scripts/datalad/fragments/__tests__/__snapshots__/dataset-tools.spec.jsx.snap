// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetTools component renders tools for public datasets 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`DatasetTools component renders tools for public snapshots 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`DatasetTools component renders tools with common props 1`] = `
<Route>
  <Component />
</Route>
`;
